### PR TITLE
Pull Request for Adding Content to C#

### PR DESCRIPTION
**Pull request by d-glennon**
### Description
Hi there! I've created content for the Array.Copy() as part of my Git learning experience on Codecademy. 
Having been working on the C# coding , I thought this would be a good first attempt at navigating the GitHub world in real life!
Personally, I'm a little unsure on how to reference Microsoft's .NET platform in this but I've done what I could. This was quite challenging and I'm sure my inexperience shows but I welcome any feedback or suggestions. Thanks! D
### Type of Change
- Adding a new entry
### Checklist
- [ x] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: .vs/VSWorkspaceState.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (SSwiniarski) [inline .vs/VSWorkspaceState.json:1]: The following files must be removed from this PR:

.vs/VSWorkspaceState.json
.vs/docs/FileContentIndex/426e5a15-bb6a-42d5-a75e-f8f1f1c0d89b.vsidx
.vs/docs/FileContentIndex/4344c5be-b400-4bd6-b8a5-e31f8673206a.vsidx
.vs/docs/FileContentIndex/71177040-e0e0-4bdf-b4d1-74cf433ef051.vsidx
.vs/docs/FileContentIndex/ecf4d5c5-c83d-4472-9b5a-7eb80d2ed8a7.vsidx
.vs/docs/FileContentIndex/read.lock
.vs/docs/v17/.wsuo
.vs/docs/v17/workspaceFileList.bin
.vs/slnx.sqlite

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+The **`.Copy()`** method in [C#](https://www.codecademy.com/resources/docs/c-sharp) 
+copies a range of elements from one array to another array. 
+
+### Syntax
+
+```
+//Copies sourceArray to destinationArray
+Array.Copy(sourceArray, destinationArray, int length);
+```
+`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`,
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:23]:
```suggestion

`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`, 
```
insert line break after coe block.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  - 'learn-c-sharp'
+  - 'paths/computer-science'
+---
+
+The **`.Copy()`** method in [C#](https://www.codecademy.com/resources/docs/c-sharp) 
+copies a range of elements from one array to another array. 
+
+### Syntax
+
+```
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:19]: This should be a ` ```pseudo ` code block.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+//Copies sourceArray to destinationArray
+Array.Copy(sourceArray, destinationArray, int length);
+```
+`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`, 
+`destinationArray` and `length`, which is specified as an integer.
+The `sourceArray` is copied, starting from the first element of the array,
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:25]:
```suggestion

The `sourceArray` is copied, starting from the first element of the array, 
```

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+### Syntax
+
+```
+//Copies sourceArray to destinationArray
+Array.Copy(sourceArray, destinationArray, int length);
+```
+`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`, 
+`destinationArray` and `length`, which is specified as an integer.
+The `sourceArray` is copied, starting from the first element of the array, 
+to the `destinationArray` which also starts from its first element. It is possible to _overload_
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:26]: Remove the link to off-site documentation.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  - 'Arrays'
+CatalogContent:
+  - 'learn-c-sharp'
+  - 'paths/computer-science'
+---
+
+The **`.Copy()`** method in [C#](https://www.codecademy.com/resources/docs/c-sharp) 
+copies a range of elements from one array to another array. 
+
+### Syntax
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:17]:
```suggestion
## Syntax
```

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+Array.Copy(sourceArray, destinationArray, int length);
+```
+`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`, 
+`destinationArray` and `length`, which is specified as an integer.
+The `sourceArray` is copied, starting from the first element of the array, 
+to the `destinationArray` which also starts from its first element. It is possible to _overload_
+this method with multiple parameters and you can learn more on the
+on Microsoft's online [.NET](https://learn.microsoft.com/en-us/dotnet/api/system.array.copy?view=net-7.0#code-try-0) 
+documentation.
+### Example
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:30]:
```suggestion

## Example

```
h2 and line break before and after header.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+The following example initializes the array `sourceArray` and the array
+`destinationArray`. Using a `foreach` loop,
+the contents of `destinationArray` are printed
+to the console:
+```
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:35]: Needs line break before and should be a ` ```cs ` code block.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+using System;
+  public class Example
+  {
+    public static void Main(string[] args)
+    {     
+       string[] sourceArray = new string[] { "red", "orange", "yellow", "green", "blue", "indigo", "violet"};
+       
+       string[] destinationArray = new string[sourceArray.Length];
+
+                                    //sourceArray.Length returns the integer for the third parameter, length.
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:45]:
```suggestion
    //sourceArray.Length returns the integer for the third parameter, length. 
```

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    
+          foreach (string element in destinationArray)
+            {
+                Console.Write($"{element} ");     
+            } 
+    }
+  }
+```
+This is what is printed to the console:
+```
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:57]: This should be a ` ```shell ` block.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+red orange yellow green blue indigo violet
+```
+
+
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:61]:
```suggestion
```
remove extra line break.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+### Codebyte Example
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:62]:
```suggestion
## Codebyte Example

```
h2 and line break after header.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+The following codebyte is runnable and shows one of the several
+overload functions with the ``Array.Copy()`` method. In this example,
+`sourceArray` is copied starting from the third element in the array **[2]**
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:65]:
```suggestion
`sourceArray` is copied starting from the third element in the array `[2]`
```

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+and stores elements **[2]**, **[3]**, **[4]** starting as the first element in `destinationArray`.
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:66]:
```suggestion
and stores elements `[2]`, `[3]`, `[4]` starting as the first element in `destinationArray`.
```

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+The remaining elements return **0**:
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:67]:
```suggestion
The remaining elements return `0`:  

```
Also line break before code block.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+
+### Codebyte Example
+The following codebyte is runnable and shows one of the several
+overload functions with the ``Array.Copy()`` method. In this example,
+`sourceArray` is copied starting from the third element in the array **[2]**
+and stores elements **[2]**, **[3]**, **[4]** starting as the first element in `destinationArray`.
+The remaining elements return **0**:  
+```
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:68]: To be a codebyte, it needs to be prefaced with ` ```codebyte/csharp `

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+       Array.Copy(sourceArray, destinationArray, sourceArray.Length);
+    
+          foreach (string element in destinationArray)
+            {
+                Console.Write($"{element} ");     
+            } 
+    }
+  }
+```
+This is what is printed to the console:
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:56]:
```suggestion

This is what is printed to the console:

```
Line breaks before and after code blocks.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+       Array.Copy(sourceArray, 2, destinationArray, 0, 3);
+    
+          foreach (int element in destinationArray)
+            {
+                Console.Write($"{element} ");     
+            } 
+    }
+  }
+```
+This is what is printed to the console:
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:88]:
```suggestion

This is what is printed to the console:

```
Line breaks before and after code blocks.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+```
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:89]: This should be a ` ```shell ` block.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@d-glennon I did a first review and left a series of comments. Most are formatting issues.

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+//Copies sourceArray to destinationArray
+Array.Copy(sourceArray, destinationArray, int length);
+```
+
+`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`, 
+`destinationArray` and `length`, which is specified as an integer.
+
+The `sourceArray` is copied, starting from the first element of the array, 
+to the `destinationArray` which also starts from its first element. It is possible to _overload_
+this method with multiple parameters and you can learn
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:29]:
```suggestion
this method with multiple parameters.
```

File: content/c-sharp/concepts/arrays/terms/copy/copy.md [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+Array.Copy(sourceArray, destinationArray, int length);
+```
+
+`.Copy()` is a static method of the `Array` object. It takes the arguments `sourceArray`, 
+`destinationArray` and `length`, which is specified as an integer.
+
+The `sourceArray` is copied, starting from the first element of the array, 
+to the `destinationArray` which also starts from its first element. It is possible to _overload_
+this method with multiple parameters and you can learn 
+more from Microsoft's online .NET documentation.
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/arrays/terms/copy/copy.md:30]:
```suggestion
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@d-glennon I had a couple more comments.

**Review by SSwiniarski — APPROVED** [review]
@d-glennon, looks good for a second review

**Review by yangc95 — APPROVED** [review]
Thanks for your patience! I made the appropriate additions to speed up this PR. Looks good to merge now.
See https://github.com/Codecademy/docs/pull/1761/commits/ceabe24eeeb8d7018d01154212eb3680c9ec3c06 for specific changes that were made. TY

**Comment by yangc95** [comment]
@d-glennon here's the live link: https://www.codecademy.com/resources/docs/c-sharp/arrays/copy

Screenshot 2023-03-19 at 8 25 41 PM

> It looks as if you're anonymous. If you want to be credited follow the instructions below to add your GitHub username to your Codecademy profile. If not, there is nothing more you need to do.
> 
> To add your GitHub username to your Codecademy profile:
> 
> 1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:
> 
> image
> 
> 2.) Then log out and log back into Codecademy using your GitHub login:
> 
> githublogin
>
> If the steps above do not work, it may be due to an email or local configuration issue.
>
> In addition, we may reach out in the future if any adjustments to these entries are required.

cc: @SSwiniarski @KTom101

**Pull request merged**